src/cli/pack_command.zig

```diff
@@ -2189,9 +2234,26 @@ pub const PackCommand = struct {
 
             return .{
                 .glob = buf[0..end],
-                .rel_path = has_leading_or_middle_slash,
-                .@"leading **/" = @"has leading **/, (could start with '!')",
-                .dirs_only = has_trailing_slash,
+                .flags = .{
+                    .rel_path = has_leading_or_middle_slash,
+                    .@"leading **/" = @"has leading **/, (could start with '!')",
+                    .dirs_only = has_trailing_slash,
+                    .negated = add_negate,
+                },
+            };
+        }
+
+        /// Invert a negated pattern to a positive pattern
+        pub fn asPositive(this: *const Pattern) Pattern {
+            bun.assertWithLocation(this.flags.negated and this.glob.len > 0, @src());
+            return Pattern{
+                .glob = this.glob[1..], // remove the leading `!`
```

Comment:
If `asPositive` is called on a pattern, `deinit` will free the wrong pointer due to `.glob = this.glob[1..]`

Reply:
This is true; pattern doesn't carry around an ownership flag. Should we use `CowSlice`?

Reply:
Fixed by using `CowString`